xbmc/cores/VideoPlayer/VideoRenderers/RenderManager.cpp

```diff
@@ -828,8 +828,15 @@ void CRenderManager::FlipPage(volatile std::atomic_bool& bStop, double pts /* =
         presentmethod = PRESENT_METHOD_BOB;
         invert = true;
       }
+      else if (interlacemethod == VS_INTERLACEMETHOD_IMX_ADVMOTION_DOUBLE)
```

Comment:
the current implementation of AutoInterlaceMethod is maximum nonsense becuase it has nothing to do with an auto-setting. don't touch this.
what you actually want to achive is having RenderManager do a single or double pass depending on deinterlacing method. why don't you ask renderer for the desird method?